thrall/app/lib/kinesis/ThrallEventConsumer.scala

```
@@ -48,26 +49,30 @@ class ThrallEventConsumer(es: ElasticSearchVersion,
           implicit val unr = Json.reads[UsageNotice]
           implicit val umr = Json.reads[UpdateMessage]
 
-          val updateMessage = Json.parse(message).as[UpdateMessage] // TODO validation
-          val timestamp = r.getApproximateArrivalTimestamp
-
-          val idForLogging = Seq(updateMessage.id, updateMessage.image.map(_.id)).flatten
-          val messageLogMessage = "(" + timestamp + "): " + updateMessage.subject + "/" + idForLogging.mkString(" ")
-          Logger.info("Got update message: " + messageLogMessage)
-
-          messageProcessor.chooseProcessor(updateMessage).map { p =>
-            val ThirtySeconds = Duration(30, SECONDS)
-            val eventuallyAppliedUpdate: Future[Any] = p.apply(updateMessage)
-            eventuallyAppliedUpdate.map { _ =>
-              Logger.info("Completed processing of update message: " + messageLogMessage)
-            }.recover {
-              case e: Throwable =>
-                Logger.error("Failed to process update message; message will be ignored: " + ("Got update message: " + messageLogMessage), e)
-            }
+          val maybeUpdateMessage = Json.parse(message).validate[UpdateMessage] <| logParseErrors |> (_.asOpt)
 
-            Await.ready(eventuallyAppliedUpdate, ThirtySeconds)
-          }
+          maybeUpdateMessage match {
```

fail fast - validate incoming message against `UpdateMessage`